### PR TITLE
add podMonitor

### PR DESCRIPTION
**Pull request by mweibel**
adds podMonitor to helm chart

### CODE REVIEW

**Review by llamerada-jp — CHANGES_REQUESTED** [review]
Thank you for your PR.
Would you please add a note that encourages users to check this feature when enabling it because I have no plan to test it in CI?

**Comment by mweibel** [comment]
sure, should I adjust the CHANGELOG.md of the chart for this?

**Comment by llamerada-jp** [comment]
@mweibel 
Hi,

It is written in the description of value looks better. This is because new users may not look at the changelogs.
And would you update `chart/pvc-autoresizer/README.md` like the below link?
https://github.com/topolvm/pvc-autoresizer/blob/main/charts/pvc-autoresizer/README.md#update-readme

**Comment by mweibel** [comment]
@llamerada-jp updated the PR with the changes. Please review again.

File: charts/pvc-autoresizer/values.yaml [hunk 1]
```diff
@@ -43,3 +43,18 @@ controller:
 
   # controller.nodeSelector -- Map of key-value pairs for scheduling pods on specific nodes.
   nodeSelector: {}
+
+# deploy a ServiceMonitor. This is not tested in CI so make sure to test it yourself.
```
Comment (llamerada-jp) [inline charts/pvc-autoresizer/values.yaml:47]: This comment is not output in README.md. Would you please adjust it?

File: charts/pvc-autoresizer/values.yaml [hunk 2]
```diff
@@ -43,3 +43,18 @@ controller:
...
+  # serviceMonitor.enabled -- If true, creates a Prometheus Operator ServiceMonitor.
+  enabled: false
+  # serviceMonitor.interval -- Interval that Prometheus scrapes metrics.
+  interval: 15s
+  # serviceMonitor.scrapeTimeout -- The timeout after which the scrape is ended
+  scrapeTimeout: 30s
+  # serviceMonitor.namespace -- Namespace which Prometheus is running in.
+  namespace: ""
+  ## [Prometheus Selector Label](https://github.com/helm/charts/tree/master/stable/prometheus-operator#prometheus-operator-1)
+  ## [Kube Prometheus Selector Label](https://github.com/helm/charts/tree/master/stable/prometheus-operator#exporters)
```
Comment (llamerada-jp) [inline charts/pvc-autoresizer/values.yaml:58]: Ditto

**Comment by mweibel** [comment]
@llamerada-jp adjusted..

**Review by llamerada-jp — COMMENTED** [review]
Is this setting for pod scraping? I read [the official document](https://prometheus-operator.dev/docs/operator/api/#monitoring.coreos.com/v1.ServiceMonitor) but I couldn't check it. The current helm chart of pvc-autoresizer doesn't generate service or endpoint resources. Do you know a good document about it?

**Comment by mweibel** [comment]
https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/user-guides/getting-started.md#deploying-a-sample-application

does that help to explain it?

**Comment by llamerada-jp** [comment]
@mweibel 
Thank you for your reply. That document is helpful. According to that document, The [`PodMonitors`](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/user-guides/getting-started.md#using-podmonitors) resource looks suitable in this situation because the service resource does not exist for the pvc-autoresizer's pod. Would you please check it?

**Comment by mweibel** [comment]
uh, that's a little embarassing :flushed: 

you're of course right - fixed!

**Comment by mweibel** [comment]
I can't currently verify this though, the PodMonitor does not get picked up for some reason. Checking and will report back when this works.

**Comment by mweibel** [comment]
regenerated the docs, too.

**Review by llamerada-jp — APPROVED** [review]
LGTM

**Review by daichimukai — APPROVED** [review]

**Pull request merged**